### PR TITLE
Release 0.1.1

### LINKED ISSUE #6: Basic mapget cache and service libraries
To interface with mapget, clients should use the `mapget::cache` API (in remote mode or as a C++ library). As a first step, we will facilitate a basic API which supports only the `Cache::tiles()` function (or the `/GET tiles` REST API). From the following UML, the `Application`, `Cache`, `DataSourceConnection` and `Configuration` classes will be implemented.

image

### PR DESCRIPTION
**Pull request by josephbirkner**
Closes #6 

Adds the `service` and `http-service` libraries, as well as a `RemoteDataSource` in `http-datasource/datasource-client.h`.

### CODE REVIEW

**Comment by josephbirkner** [comment]
Missing:

- [ ] `HttpService` in `httpservice` library
- [ ] `RemoteDataSource` in `httpdatasource/datasourceclient.h`
- [ ] Tests
- [ ] Example
- [ ] Update docs

File: examples/cpp/http-datasource/main.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include "mapget/http-datasource/datasource-server.h"
+
+using namespace mapget;
+using json = nlohmann::json;
+namespace fs = std::filesystem;
+
+// Class to encapsulate all logic of our data source
+class MyRemoteDataSource
```
Comment (MisterGC) [inline examples/cpp/http-datasource/main.cpp:12]: Is it a necessity to use composition for remote data sources? Why does the local one use inheritance and the remote one not?

**Review by MisterGC — APPROVED** [review]
Successfully tested the examples (Linux, GCC 11.3):
- local data source by just executing it and checking the output
- http datasource with minimal service by invoking status, sources and using the tiles post request:
 
``` bash
curl -X POST \
    -H "Content-Type: application/json" \
    -H "Accept: application/jsonl" \
    -H "Connection: close" \
    -d '{
    "requests": [
       {
           "mapId": "MyFancyMap",
           "layerId": "MyFancyLayer",
           "tileIds": [1, 2, 3]
       }
    ]
}' "http://localhost:8080/tiles"
```

Regarding using curl: 

- [ ]  It would be good to have a example in the README which is aligned to the examples that I can run, so that I can directly copy the curl call to execute the test
- [ ] Is it expected that the curl hangs/times-out if I use an unknown mapId?

File: libs/http-service/src/cli.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    std::string server, map, layer;
+    std::vector<int> tiles;
+    fetch_cmd->add_option("-s,--server", server, "Server to connect to in format <host:port>.")->required();
+    fetch_cmd->add_option("-m,--map", map, "Map to retrieve.")->required();
+    fetch_cmd->add_option("-l,--layer", layer, "Layer of the map to retrieve.")->required();
+    fetch_cmd->add_option("-t,--tile", tiles, "Tile of the map to retrieve. Can be specified multiple times.")->required();
+    fetch_cmd->callback([&]() { fetch(server, map, layer, tiles); });
+
+    try {
+        std::reverse(args.begin(), args.end());
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply libs/http-service/src/cli.cpp:118]: This is expected by CLI11, says so in the docstring. I was also surprised 😄

**Pull request merged**